source/common/network/raw_buffer_socket.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/transport_socket.h"
+
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Network {
+
+class RawBufferSocket : public TransportSocket, protected Logger::Loggable<Logger::Id::connection> {
+public:
+  // Network::TransportSocket
+  void setTransportSocketCallbacks(TransportSocketCallbacks& callbacks) override;
+  std::string protocol() const override;
+  bool canFlushClose() override { return true; }
+  void closeSocket(Network::ConnectionEvent) override {}
```

Comment:
Doesn't this need to close the socket?

Reply:
No, the raw buffer socket doesn't have to do anything other than the one does in ConnectionImpl.